### ISSUE TITLE
No significant interactions found

### ISSUE THREAD

**Issue by BoxWong** [ay-lab/FitHiChIP#30]
Hi there,

I've encountered a problem trying to get FitHiChIP to run.
I checked the log file and found these warnings:
_1: In data.table::fread(temp_peakfile, header = F, sep = "\t", stringsAsFactors = F) :
  File '/usr/users/xwang6/work/hichip/hg19/fithichip/k180_siCont1_results/NormFeatures/temp_peaks.bed' has size 0. Returning a NULL data.frame.
2: In data.table::fread(temp_peakfile, header = F, sep = "\t", stringsAsFactors = F) :
  File '/usr/users/xwang6/work/hichip/hg19/fithichip/k180_siCont1_results/NormFeatures/temp_peaks.bed' has size 0. Returning a NULL data.frame._

I went to that directory and indeed I didn't find any files named "temp_peaks.bed". Any idea how this would affect the result?

Another problem is that at the end of the log file, I found some errors as follows:
 Error in as.data.frame.default(x[[i]], optional = TRUE, stringsAsFactors = stringsAsFactors) :
  cannot coerce class ‘"try-error"’ to a data.frame
Calls: as.data.frame ... <Anonymous> -> as.data.frame -> as.data.frame.default
In addition: Warning message:
In parallel:::mclapply(1:length(avg_int_dist), mc.cores = ncore,  :
  all scheduled cores encountered errors in user code
Execution halted
******** FINISHED calling significant interactions
ERROR !!!!!!!! FitHiChIP could not compute the statistical significance of input interactions
Check the input parameters, or check if the number of input nonzero contact locus pairs are too few !!!
If you are using peak to peak background only (UseP2PBackgrnd=1 or stringent background), check the number of nonzero peak-to-peak locus pairs (peak-to-peak) !!! In such a case, you sho
awk: fatal: cannot open file `/usr/users/xwang6/work/hichip/hg19/fithichip/k180_siCont1_results/FitHiChIP_Peak2ALL_b2000_L2000_U2000000/P2PBckgr_0/ICE_Bias/FitHiC_BiasCorr/FitHiChIP_k18
----- Extracted significant interactions ---- FDR threshold lower than: 0.01
SORRY !!!!!!!! FitHiChIP could not find any statistically significant interactions
Check the input parameters, or check if the number of input nonzero contact locus pairs are too few !!!
If you are using peak to peak background only (UseP2PBackgrnd=1 or stringent background), check the number of nonzero peak-to-peak locus pairs (peak-to-peak) !!! In such a case, you sho
There is no significant interaction - so no WashU specific session file is created !!
Updated CurrIntType: 4

 **** Now summarizing FitHiChIP results in the HTML file ***

***** There are one or more errors / notifications (such as zero interaction count) associated with this execution !!!
***** Please check the console to know about them !!

Are these errors caused by the warnings I mentioned previously? 

Thanks in advance!

Best,
Xin

**Comment by ay-lab** [comment]
Hi @BoxWong ,
Thanks for your query. 
1) First, please clone the latest GitHub repository (updated today). 
2) Then, please check if your input valid pairs file, and also input peak file has chromosome information as chr1, chr2, ... and not 1, 2, ...
3) Your input peak file is bed formatted, i.e. first field is chromosome name, second and third fields are peak intervals.
I think, parsing of peak file encountered an error, resulting subsequent error messages. Please check and reply me for any issues.

**Comment by BoxWong** [comment]
Hi, 
Thanks for your reply and I've tried what you suggested. 

Re 1: I am using the latest version now.
Re 2 and 3: I checked the input valid pairs file and the input peak file. Both of them looked fine to me:
an example of my input pairs:
_K00203:352:HFTJGBBXY:8:1122:12510:47999/2       chr1    13315   +       chr3    152616442       -       1907    HIC_chr1_7      HIC_chr3_360988 35      42
K00203:352:HFTJGBBXY:8:1113:20973:17438/1       chr1    20103   +       chr3    53499570        -       899     HIC_chr1_22     HIC_chr3_134277 35      42
K00203:352:HFTJGBBXY:8:2222:16833:30345/1       chr1    20103   +       chr7    25430426        +       452     HIC_chr1_22     HIC_chr7_62192  35      42
K00203:352:HFTJGBBXY:8:1120:32106:44570/2       chr1    20136   -       chr7    55545886        -       618     HIC_chr1_22     HIC_chr7_135016 32      42
K00203:352:HFTJGBBXY:8:1201:19664:11231/2       chr1    51435   +       chr15   99578398        -       459     HIC_chr1_96     HIC_chr15_203076        35      42
K00203:352:HFTJGBBXY:8:2204:15970:6730/2        chr1    51659   +       chr8    18932194        +       384     HIC_chr1_99     HIC_chr8_46047  31      35
K00203:352:HFTJGBBXY:8:1225:5568:18880/2        chr1    54379   +       chr9    120186317       -       512     HIC_chr1_104    HIC_chr9_244916 35      42
K00203:352:HFTJGBBXY:8:1227:13667:40684/1       chr1    55509   +       chr19   791610  +       1293    HIC_chr1_109    HIC_chr19_2086  39      42
K00203:352:HFTJGBBXY:8:1103:25093:18775/1       chr1    56277   -       chr7    135701704       -       826     HIC_chr1_109    HIC_chr7_330274 35      42
K00203:352:HFTJGBBXY:8:1125:9191:13763/2        chr1    56983   +       chr1    168443185       -       391     HIC_chr1_111    HIC_chr1_377014 42      42_

an example of my input peak file:
_chr1    28578   29987
chr1    713346  714049
chr1    714195  715678
chr1    762034  763443
chr1    839713  840041
chr1    840228  840437
chr1    893951  894631
chr1    895650  896677
chr1    901878  902395
chr1    911523  913463_

So, they look fine to me, what do you think about it?
I tried using the coverage bias correction and it worked. But it only gave me 12 significant interactions (from _FitHiChIP.interactions_FitHiC_Q0.05.bed_) with FDR < 0.05 (out of 264537 interactions from _FitHiChIP.interactions_FitHiC.bed_). One thing might be worth noticing is that I still  encountered this warning:
_Warning message:
In data.table::fread(temp_peakfile, header = F, sep = "\t", stringsAsFactors = F) :
  File '/usr/users/xwang6/work/hichip/fithichip/l26_q0.05/NormFeatures/temp_peaks.bed' has size 0. Returning a NULL data.frame._
which I've also encountered before as I have described in the first issue report. But at the end of the log file it told me that the Pipeline is completely executed and congratulations and so on. So I guess this is fine, right? Just can't believe that I only got 12 significant interactions. 

As for the ICE bias correction, I still have the same problem. The same warning message:
 _Warning message:
In data.table::fread(temp_peakfile, header = F, sep = "\t", stringsAsFactors = F) :
  File '/usr/users/xwang6/work/hichip/fithichip/l26_ICE_q0.05/NormFeatures/temp_peaks.bed' has size 0. Returning a NULL data.frame._

and the end looks like this:

_Error in as.data.frame.default(x[[i]], optional = TRUE, stringsAsFactors = stringsAsFactors) :
  cannot coerce class ‘"try-error"’ to a data.frame
Calls: as.data.frame ... <Anonymous> -> as.data.frame -> as.data.frame.default
In addition: Warning message:
In parallel:::mclapply(1:length(avg_int_dist), mc.cores = ncore,  :
  all scheduled cores encountered errors in user code
Execution halted
******** FINISHED calling significant interactions
ERROR !!!!!!!! FitHiChIP could not compute the statistical significance of input interactions
Check the input parameters, or check if the number of input nonzero contact locus pairs are too few !!!
If you are using peak to peak background only (UseP2PBackgrnd=1 or stringent background), check the number of nonzero peak-to-peak locus pairs (peak-to-peak) !!! In such a case, you should go for loose ba
awk: fatal: cannot open file `/usr/users/xwang6/work/hichip/fithichip/l36_ICE_q0.05/FitHiChIP_Peak2ALL_b2000_L10000_U5000000/P2PBckgr_0/ICE_Bias/FitHiC_BiasCorr/FitHiChIP.interactions_FitHiC.bed' for read
----- Extracted significant interactions ---- FDR threshold lower than: 0.05
SORRY !!!!!!!! FitHiChIP could not find any statistically significant interactions
Check the input parameters, or check if the number of input nonzero contact locus pairs are too few !!!
If you are using peak to peak background only (UseP2PBackgrnd=1 or stringent background), check the number of nonzero peak-to-peak locus pairs (peak-to-peak) !!! In such a case, you should go for loose ba
There is no significant interaction - so no WashU specific session file is created !!
Updated CurrIntType: 4

 **** Now summarizing FitHiChIP results in the HTML file ***

***** There are one or more errors / notifications (such as zero interaction count) associated with this execution !!!
***** Please check the console to know about them !!_


I would appreciate if you can help me out!
Thanks a lot!

Best regards,
Xin

**Comment by BoxWong** [comment]
Sorry, me again. 
I was just reading the error message from the end of the log file again. I was thinking maybe it is indeed the case that I have too few interactions? Probably this also explains why I had only 12 significant interactions from this coverage correction. 
But I still don't get why I receive the following warning message:
_In data.table::fread(temp_peakfile, header = F, sep = "\t", stringsAsFactors = F) :
  File '/usr/users/xwang6/work/hichip/fithichip/l26_ICE_q0.05/NormFeatures/temp_peaks.bed' has size 0. Returning a NULL data.frame._

Best regards,
Xin

**Comment by ay-lab** [comment]
Hi @BoxWong 
FitHiChIP checks peaks for individual chromosomes, writes them in temporary file, and processes. Probably your input peak file does not contain any peaks for one or more chromosomes. That's why the warning message comes. The output is not affected by this warning. But, of course, I'll fix this issue in the coming version.

I would suggest using coverage bias + peak to all background (P2P = 0) for calling significant interactions, if you haven't already done so.
